src/controller/OperationalCredentialsDelegate.h

```diff
@@ -50,17 +50,15 @@ class DLL_EXPORT OperationalCredentialsDelegate
      *
      *   The delegate will call `onCompletion` when the NOC certificate chain is ready.
      *
-     * @param[in] csrElements          CSR elements as per specifications section 11.22.5.6. NOCSR Elements.
-     * @param[in] attestationSignature Attestation signature as per specifications section 11.22.7.6. CSRResponse Command.
-     * @param[in] DAC                  Device attestation certificate received from the device being commissioned
-     * @param[in] PAI                  Product Attestation Intermediate certificate
-     * @param[in] PAA                  Product Attestation Authority certificate
-     * @param[in] onCompletion         Callback handler to provide generated NOC chain to the caller of GenerateNOCChain()
+     * @param[in] csr          Certificate Signing Request
+     * @param[in] DAC          Device attestation certificate received from the device being commissioned
+     * @param[in] PAI          Product Attestation Intermediate certificate
+     * @param[in] PAA          Product Attestation Authority certificate
+     * @param[in] onCompletion Callback handler to provide generated NOC chain to the caller of GenerateNOCChain()
      *
      * @return CHIP_ERROR CHIP_NO_ERROR on success, or corresponding error code.
      */
-    virtual CHIP_ERROR GenerateNOCChain(const ByteSpan & csrElements, const ByteSpan & attestationSignature, const ByteSpan & DAC,
-                                        const ByteSpan & PAI, const ByteSpan & PAA,
+    virtual CHIP_ERROR GenerateNOCChain(const ByteSpan & csr, const ByteSpan & DAC, const ByteSpan & PAI, const ByteSpan & PAA,
                                         Callback::Callback<OnNOCChainGeneration> * onCompletion) = 0;
```

Comment:
Although it failed to succeed in this, this interface was previously attempting to pass enough information to the OperationalCredentialsDelegate to be able to verify the attestation signature for the CSR remotely in the signer.

This PR completely removes this.  But what are we trying to accomplish?  I think we would like the standard interfaces to support remote verification of this signature because this is sort of the point.  If the local node has verified attestation already (as will be the case with the sdk's current behavior), it doesn't need to verify attestation of the CSR at all because this is redundant: it already will have performed attestation and has maintained the same PASE session since then.

For this to actually be useful then, OperationalCredentialsDelegate needs to generate the CSR nonce, and then needs to be passed enough context with the CSR to verify the signature.  This would look something like:

```
CHIP_ERROR OperationalCredentialsDelegate::GetCSRNonce(void * context, ByteSpan & nonce);
CHIP_ERROR OperationalCredentialsDelegate::GenerateNOCChain(void *context, ByteSpan csrElements,
    ByteSpan challenge, ByteSpan signature, ByteSpan pai, ByteSpan dac);
```

Reply:
+1

Reply:
attestationSignature and csrElements parameters are removed as they are already being validated in the VerifyNodeOperationalCSRInformation method. Having these parameters sent again into the GenerateNOCChain method will make it redundant.

Reply:
Yes, but the point is for the signer to be able to validate the attestation signature itself.  As explained, calling VerifyNodeOperationalCSRInformation locally provides little value if the local node has also verified attestation and maintained a PASE session the entire time.  For the node that has established PASE, the CSR is already implicitly trustable by virtue of attestation and the fact the CSR has been communicated across the PASE channel.

In contrast, a remote certificate signer cannot verify the provenance of the CSR unless it generates the nonce itself and verifies the signature itself.

Without this, a remote signer is forced to trust the node that has established PASE with the commissionee to vouch for attestation.  This may not be acceptable for many signers.

Reply:
So for the purposes of moving this PR forward, reverting the GenerateNOCChain interface changes probably makes sense.

But just to be clear, this does not fix the problem I've laid out.  Remote verification of the CSR signature in the signer is a valid and important use case.  The code didn't support this before, and no version of this PR so far has allowed for such a thing.  Maybe we can open a separate issue.

Reply:
I agree, we can get this PR merged and track this open issue separately.

Reply:
By the way, this is readily addressable in commissioner implementations that adopt the approach introduced in https://github.com/project-chip/connectedhomeip/pull/13294.

See: https://github.com/project-chip/connectedhomeip/blob/7182126ee4e6b6de4516d01b321f67a4915b0326/src/commissioner/ExampleCommissioningStateMachine.h#L235